Python/chapter02/2.4 - Partition/miguel_2.4_soln.py

```diff
@@ -0,0 +1,241 @@
+"""
+Python version 3.7.0
+2.4 - Partition
+Write code to partition a linked list around a value x,
+such that all nodes less than x come before all nodes
+greater than or equal to x. If x is contained within
+the list, the values of x only need to be after
+the elements less than x. The partition element x
+can appear anywhere in the 'right partition'; it does
+not need to appear between the left and right
+partitions.
+EXAMPLE
+Input:  3 -> 5 -> 8 -> 5 -> 10 -> 2 -> 1 [partition = 5]
```

Comment:
in the context of Quicksort, the value x is normally called ["pivot"](https://en.wikipedia.org/wiki/Quicksort). we cannot call it "partition" since that's the name of the operation as a whole, so might as well call it "pivot". Same in the rest of the file.

Reply:
changed